odoo/addons/base/tests/common.py

```diff
@@ -251,15 +254,23 @@ def login(self, user, password):
 
         self.testing_smtp_session = TestingSMTPSession()
 
+        get_origin = config.get
         IrMailServer = self.env['ir.mail_server']
-        connect = IrMailServer.connect
-        find_mail_server = IrMailServer._find_mail_server
+        connect_origin = IrMailServer.connect
+        find_mail_server_origin = IrMailServer._find_mail_server
+
+        def config_get(*args, **kwargs):
+            if args[0] == 'smtp_server':
+                return self.config_smtp_server
+            return get_origin(*args, **kwargs)
```

Comment:
isn't it better to return `None` ? so we simulate the fact that no config is configured in the odoo-bin arguments, and we don't need the argument `config_smtp_server`

Reply:
It could have a side effect on other stuff... there are a lot of "default configuration" elements, I would keep the mock as small as possible.